libs/core/src/lib/title/title.component.ts

```diff
@@ -25,8 +25,19 @@ export type HeaderSizes = 1 | 2 | 3 | 4 | 5 | 6;
 })
 export class TitleComponent implements OnInit, DoCheck {
     /** The size of the header */
+    _headerSize: HeaderSizes = null;
+
+    get headerSize(): HeaderSizes {
+        return this._headerSize;
+    }
+
     @Input()
-    headerSize: HeaderSizes = null;
+    set headerSize(value: HeaderSizes) {
+        this._headerSize = value;
+        if (this._appliedHeaderSize !== this.headerSize) {
+            this._setHeaderSize();
```

Comment:
Inside _setHeaderSize method, do we really need to use classList? Why not a template-driven approach? Should be much easier and more obvious

Reply:
I don't know why its not template-driven, I guess it was discussed and selected on feature creation step , this is PR related to fix, I guess if we want to change structure of the component - we need make some change request issue, it will affect to other part of the project, other components, to big change for a fix. Definitely should ne created separated issue.

Reply:
is there a need to have doCheck hook If we start relying on an input setter?